pkg/receive/handler.go

```diff
@@ -607,68 +644,41 @@ func (h *Handler) fanoutForward(pctx context.Context, tenant string, wreqs map[e
 		tLogger = log.With(h.logger, logTags)
 	}
 
-	ec := make(chan error)
+	ec := make(chan writeResponse)
```

Comment:
Could `ec` receive a better name? It's used many times in the next hundreds of lines and the name isn't clear. Suggestions: `errorChannel`, if that's even what is actually is. 😅